File: Domain/Waiting.cs
using System;
using System.ComponentModel.DataAnnotations;

namespace Domain
{
    public class Waiting
    {
        [Key]
        public Guid id { get; set; }
        public string UserName { get; set; }
        public string DisplayName { get; set; }
        public string PhoneNumber { get; set; }
        public string BarberName { get; set; }
        public string Date { get; set; }

    }
}

File: web-app/src/models/waiting.ts
export interface Waiting {
    id: string;
    userName: string;
    displayName: string;
    phoneNumber: string;
    barberName: string;
    date: string;

}

export class Waiting implements Waiting {
    constructor(init?: WaitingFormValues) {
        Object.assign(this, init);
    }
}

export class WaitingFormValues {

    id?: string
    userName?: string;
    displayName?: string;
    phoneNumber?: string;
    barberName?: string;
    date?: string;

    constructor(waiting?: WaitingFormValues) {
        if (waiting) {
            this.id = waiting.id;
            this.userName = waiting.userName;
            this.displayName = waiting.displayName;
            this.phoneNumber = waiting.phoneNumber;
            this.barberName = waiting.barberName;
            this.date = waiting.date;

        }
    }
}